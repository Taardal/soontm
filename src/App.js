import React from "react";
import { StatusBar, View } from "react-native";
import { Provider } from "react-redux";
import { createStackNavigator } from "react-navigation";
import store from "./reduxStore";
import GalleryContainer from "./GalleryContainer";
import Details from "./Details";
import { styles } from "./App.styles";

const Navigator = createStackNavigator(
  {
    Gallery: {
      screen: GalleryContainer
    },
    Details: {
      screen: Details
    }
  },
  {
    initialRouteName: "Gallery",
    navigationOptions: {
      headerStyle: {
        backgroundColor: styles.actionBar.backgroundColor
      },
      headerTintColor: styles.actionBar.tintColor,
      headerTitleStyle: {
        fontWeight: styles.actionBar.fontWeight
      }
    }
  }
);

const App = () => (
  <Provider store={store}>
    <View style={styles.container}>
      <StatusBar barStyle="light-content" backgroundColor={styles.statusBar.backgroundColor} />
      <Navigator />
    </View>
  </Provider>
);

export default App;
