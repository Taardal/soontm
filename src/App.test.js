import React from "react";
import ReactTestRenderer from "react-test-renderer";
import App from "./App";

jest.mock("View", () => "View");
jest.mock("StatusBar", () => "StatusBar");
jest.mock("react-navigation", () => {
  const RealComponent = require.requireActual("react-navigation");
  const React = require("React");
  class Navigator extends React.Component {
    render() {
      return React.createElement("Navigator", this.props, this.props.children);
    }
  }
  Navigator.propTypes = RealComponent.propTypes;
  return {
    createStackNavigator: () => Navigator
  };
});
jest.mock("./GalleryContainer", () => "GalleryContainer");
jest.mock("./Details", () => "Details");
jest.mock("./App.styles", () => ({
  styles: {
    container: {
      flex: 1
    },
    statusBar: {
      backgroundColor: "black"
    },
    actionBar: {
      backgroundColor: "#212121",
      tintColor: "#fff",
      fontWeight: "bold"
    }
  }
}));

describe("App", () => {
  it("renders", () => {
    const viewTree = ReactTestRenderer.create(<App />).toJSON();
    console.log(viewTree);

    expect(viewTree).toMatchSnapshot();
    expect(viewTree.type).toBe("View");
    expect(viewTree.children.length).toBe(2);
    expect(viewTree.children[0].type).toBe("StatusBar");
    expect(viewTree.children[1].type).toBe("Navigator");
  });
});
