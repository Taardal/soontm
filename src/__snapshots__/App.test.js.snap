// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <StatusBar
    backgroundColor="black"
    barStyle="light-content"
  />
  <Navigator />
</View>
`;
